core/src/main/scala/cr/pulsar/ConsumerOpts.scala

```diff
@@ -0,0 +1,21 @@
+package cr.pulsar
+
+import org.apache.pulsar.client.api.SubscriptionInitialPosition
+
+// Builder-style abstract class instead of case class to allow for bincompat-friendly extension in future versions.
+sealed abstract class ConsumerOpts {
```

Comment:
We could move this inside `Consumer` object and call it `Options`. Also we could do the same for `Producer`